### ISSUE TITLE
Add hint to disable user confirmation to SSH Agent Protocol Error

### ISSUE THREAD

**Issue by kimberlyeet** [keepassxreboot/keepassxc#9568]
## Overview
throws error when unlocking while openssh agent is enabled, agent working fine

## Steps to Reproduce
1. Enable the openSSH agent integration
2. unlock your database

## Expected Behavior
SSH agent working, no errors

## Actual Behavior
SSH agent is working fine but throws this error when unlocking my database:
![image](https://github.com/keepassxreboot/keepassxc/assets/44536631/50bcf6ff-88f4-4e53-a049-ca9ae57add14)

## Context
Using the latest version on Windows 11 22H2

KeePassXC - Version 2.7.5
Revision: 9d0537b

Operating System: Windows 11 22H2

**Comment by droidmonkey** [comment]
What version of openssh agent are you running and in what context (ie, WSL or windows feature or otherwise)?

**Comment by droidmonkey** [comment]
This error is only thrown if the SSH Agent fails to respond to the messages we send it: https://github.com/keepassxreboot/keepassxc/blob/develop/src/sshagent/SSHAgent.cpp#L190 as an example.

**Comment by kimberlyeet** [comment]
Using openssh as a windows feature.
ssh -V outputs:
`OpenSSH_for_Windows_8.6p1, LibreSSL 3.4.3`

**Comment by psztoch** [comment]
Windows users commonly use WSL now.
It would be convenient to support Pageant and local OpenSSH from Windows as well as support OpenSSH in WSL (as an independent third option).

**Comment by xboxones1** [comment]
I confirm. I also have the latest Windows 11 22h2 and keepassxc doesn't work with the OpenSSH server of Windows itself.

**Comment by xboxones1** [comment]
Works well for me on the same configuration

**Comment by vlascik** [comment]
same symptoms, same system, same error, no identities are added to the ssh-agent.

EDIT: but only when "Require user confirmation" is checked on the key entry -> SSH Agent tab, if unchecked, adding to the ssh-agent works and there's no error.

**Comment by droidmonkey** [comment]
Then that means the agent does not support that option. There is no workaround for that. 

@kimberlyeet do you still have the agent protocol error if you disable user Confirmation?

**Comment by vlascik** [comment]
In that case there should be a better error message, I only discovered the workaround by accident (and after wasting a good amount of time on finding it too).

**Comment by kimberlyeet** [comment]
> same symptoms, same system, same error, no identities are added to the ssh-agent.
> 
> EDIT: but only when "Require user confirmation" is checked on the key entry -> SSH Agent tab, if unchecked, adding to the ssh-agent works and there's no error.

this works for me too, tyvm :)
But I agree, maybe the error should be more specific.

**Comment by droidmonkey** [comment]
We cannot determine the precise error because the  ssh agent protocol is garbage. It doesn't return an error besides "error". We could add a hint that this _might be_ the problem.

**Comment by droidmonkey** [comment]
This is actually a duplicate of #4374

**Comment by aderchox** [comment]
**TL;DR:** This was really exhausting for me to finally realize that **I had to disable both options** "Require user confirmation when this key is used" and "Remove key from agent after". 😅

**More detailed:**
1. If you get this error: "no agent running cannot add identity"
Go to services.msc and start this service: "OpenSSH Authentication Agent"
Check with `ssh-add -l` to see if it's connected.

2. If you get this error: "Agent Protocol Error"
Disable "Require user confirmation when this key is used".

3. If you get this error: "Agent refused this identity"
Disable "Remove key from agent after".

Now again check with `ssh-add -l` to see if the key gets listed. If there's no error but the key isn't listed either, right click on it and add it and then check again. Also note that if your key has a passphrase, you have to enter it in the `password` field of the entry.

------------------------------------------------------------------------------
===== Updating the OpenSSH service may help either if none of the above solutions work: =====

The version of SSH client that comes with Windows 10 before version 21H1 is buggy and doesn't work with RSA keys (this has been patched in 21H1). If you are running a version of Windows earlier than 21H1, then you must apply this workaround:

    - Stop the agent service.
    - Download the latest release of [OpenSSH-Win64.zip](http://openssh-win64.zip/) from GitHub. Extract it to C:\Program Files\OpenSSH-Win64.
    - Open regedit and go to HKLM\SYSTEM\CurrentControlSet\Services\ssh-agent. Change ImagePath to C:\Program Files\OpenSSH-Win64\ssh-agent.exe
    - Start the service.

Now enable the SSH Agent in KeePassXC settings and check Use OpenSSH for Windows instead of Pageant.

Use ssh-add -l to check if your keys are loaded (if you've chosen to add them manually, you can do this by right-clicking them and selecting Add key to SSH Agent).

**Comment by erikhubers** [comment]
> .......
> ===== Updating the OpenSSH service may help either if none of the above solutions work: =====
> 
> The version of SSH client that comes with Windows 10 before version 21H1 is buggy and doesn't work with RSA keys (this has been patched in 21H1). If you are running a version of Windows earlier than 21H1, then you must apply this workaround:
> 
> ```
> - Stop the agent service.
> - Download the latest release of [OpenSSH-Win64.zip](http://openssh-win64.zip/) from GitHub. Extract it to C:\Program Files\OpenSSH-Win64.
> - Open regedit and go to HKLM\SYSTEM\CurrentControlSet\Services\ssh-agent. Change ImagePath to C:\Program Files\OpenSSH-Win64\ssh-agent.exe
> - Start the service.
> ```
> 
> Now enable the SSH Agent in KeePassXC settings and check Use OpenSSH for Windows instead of Pageant.
> 
> Use ssh-add -l to check if your keys are loaded (if you've chosen to add them manually, you can do this by right-clicking them and selecting Add key to SSH Agent).

I can confirm this works. Thanks for mentioning this solution/workaround!
